packages/shylu/shylu_dd/frosch/src/InterfaceSets/FROSch_DDInterface_def.hpp

```diff
@@ -808,115 +707,33 @@ namespace FROSch {
-                                dofsLocal[k] = DofsPerNode_*nodeIDLocal+k;
-                                dofsGlobal[k] = DofsPerNode_*nodeIDGlobal+k;
-                            }
-                            
-                            tmpEntity->addNode(nodeIDGamma,nodeIDLocal,nodeIDGlobal,DofsPerNode_,dofsGamma,dofsLocal,dofsGlobal);
-                            tmpEntity->resetEntityType(VertexType);
-                            Vertices_->addEntity(tmpEntity);
-                            
-                            break;
-                            
-                        default:
-                            
-                            sortunique(components[i]);
-                            
-                            for (UN j=0; j<components[i].size(); j++) {
-                                nodeIDGamma = componentsGamma[i][j];
-                                nodeIDLocal = components[i][j];
-                                nodeIDGlobal = NodesMap_->getGlobalElement(nodeIDLocal); //cout << "edge " << nodeIDGlobal << std::endl;
-                                for (UN k=0; k<DofsPerNode_; k++) {
-                                    dofsGamma[k] = DofsPerNode_*nodeIDGamma+k;
-                                    dofsLocal[k] = DofsPerNode_*nodeIDLocal+k;
-                                    dofsGlobal[k] = DofsPerNode_*nodeIDGlobal+k;
-                                }
-                                
-                                tmpEntity->addNode(nodeIDGamma,nodeIDLocal,nodeIDGlobal,DofsPerNode_,dofsGamma,dofsLocal,dofsGlobal);
-                            }
-                            tmpEntity->resetEntityType(EdgeType);
-                            Edges_->addEntity(tmpEntity);
-                            
-                            break;
-                    }
-                    break;
+            
+            sortunique(components[i]);
+            
+            //
+            for (UN j=0; j<components[i].size(); j++) {
+                nodeIDGamma = componentsGamma[i][j];
+                nodeIDLocal = components[i][j];
+                nodeIDGlobal = NodesMap_->getGlobalElement(nodeIDLocal);
+                for (UN k=0; k<DofsPerNode_; k++) {
+                    dofsGamma[k] = DofsPerNode_*nodeIDGamma+k;
+                    dofsLocal[k] = DofsPerNode_*nodeIDLocal+k;
+                    dofsGlobal[k] = DofsPerNode_*nodeIDGlobal+k;
+                }
+                
+                tmpEntity->addNode(nodeIDGamma,nodeIDLocal,nodeIDGlobal,DofsPerNode_,dofsGamma,dofsLocal,dofsGlobal);
             }
+            tmpEntity->resetEntityType(DefaultType);
+            EntitySetVector_[componentsMultiplicity[i]]->addEntity(tmpEntity);
         }
+
+        // Remove the empty entity stemming from the interior nodes
+        removeEmptyEntities();
         
-        // If Dimension_ == 2, then the faces are indeed edges and there are no faces
-        if (Dimension_ == 2) {
```

Comment:
Is this check no longer needed ?

Reply:
Thanks for the quick approval of the PR. While working on the graph for the interface entities, I also introduced a generalization w.r.t. faces in 3D and edges in 2D. Since faces in 3D are the same as edges in 2D for the GDSW algorithms, I was able to omit the distinction between 2D and 3D in this part of the code. This should make the implementation more general.